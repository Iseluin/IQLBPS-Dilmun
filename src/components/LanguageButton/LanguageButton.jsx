import { Fragment } from 'react';
import { Menu, Transition } from '@headlessui/react';
import { SiGoogletranslate } from 'react-icons/si';

import UsFlag from '../../assets/img/us-flag.svg';
import IraqFlag from '../../assets/img/iraq-flag.svg';

const classNames = (...classes) => {
  return classes.filter(Boolean).join(' ');
};

const LanguageButton = () => {
  const handleLanguageChange = (language) => {
    localStorage.setItem('language', language);
  };

  return (
    <Menu
      as="div"
      data-testid="language-button"
      className="relative inline-block text-left"
    >
      <div>
        <Menu.Button
          data-testid="language-botton-toggle"
          className="md:ml-6 text-2xl md:mt-0 mt-2 text-background hover:text-secondary duration-500"
        >
          <SiGoogletranslate />
        </Menu.Button>
      </div>

      <Transition
        as={Fragment}
        enter="transition ease-out duration-100"
        enterFrom="transform opacity-0 scale-95"
        enterTo="transform opacity-100 scale-100"
        leave="transition ease-in duration-75"
        leaveFrom="transform opacity-100 scale-100"
        leaveTo="transform opacity-0 scale-95"
      >
        <Menu.Items className="absolute -right-24 md:right-0 z-10 mt-2 w-56 origin-center md:origin-top-right rounded-md bg-primary shadow-lg ring-1 ring-white ring-opacity-5 focus:outline-none">
          <div>
            <Menu.Item>
              {({ active }) => (
                <div
                  data-testid="language-button-option"
                  onClick={() => handleLanguageChange('en')}
                  aria-hidden="true"
                  className={classNames(
                    active ? 'bg-secondary rounded-md' : '',
                    'text-sm flex justify-between items-center px-4 cursor-pointer text-background'
                  )}
                >
                  <h3>English</h3>
                  <img className="h-8 w-8" src={UsFlag} alt="English" />
                </div>
              )}
            </Menu.Item>
            <Menu.Item>
              {({ active }) => (
                <div
                  data-testid="language-button-option"
                  onClick={() => handleLanguageChange('ar')}
                  aria-hidden="true"
                  className={classNames(
                    active ? 'bg-secondary rounded-md' : '',
                    'text-sm flex justify-between items-center px-4 cursor-pointer text-background'
                  )}
                >
                  <h3>Arabic</h3>
                  <img className="h-8 w-8" src={IraqFlag} alt="Arabic" />
                </div>
              )}
            </Menu.Item>
          </div>
        </Menu.Items>
      </Transition>
    </Menu>
  );
};

export default LanguageButton;
