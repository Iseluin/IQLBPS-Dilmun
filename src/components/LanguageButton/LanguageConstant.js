import UsFlag from '../../assets/img/us-flag.svg';
import IraqFlag from '../../assets/img/iraq-flag.svg';

const languages = [
  {
    name: 'English',
    code: 'en',
    flag: UsFlag,
    direction: 'ltr',
  },
  {
    name: 'العربية',
    code: 'ar',
    flag: IraqFlag,
    direction: 'rtl',
  },
  {
    name: 'کوردی',
    code: 'ku',
    flag: 'https://upload.wikimedia.org/wikipedia/commons/3/35/Flag_of_Kurdistan.svg',
    direction: 'rtl',
  },
];

export default languages;
