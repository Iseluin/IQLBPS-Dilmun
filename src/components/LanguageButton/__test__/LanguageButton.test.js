import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import LanguageButton from '../LanguageButton';
import UsFlag from '../../../assets/img/us-flag.svg';
import IraqFlag from '../../../assets/img/iraq-flag.svg';

test('should render LanguageButton component ', () => {
  render(<LanguageButton />);
  const LanguageButtonElement = screen.getByTestId('language-button');
  expect(LanguageButtonElement).toBeInTheDocument();
});

test('Language Button components shloud has 2 options', async () => {
  const user = userEvent.setup();
  render(<LanguageButton />);

  await user.click(screen.getByTestId('language-botton-toggle'));

  const options = screen.getAllByTestId('language-button-option');
  expect(options).toHaveLength(2);
});

test('option 1 shloud has English title and Us Flag', async () => {
  const user = userEvent.setup();
  render(<LanguageButton />);

  await user.click(screen.getByTestId('language-botton-toggle'));

  const options = screen.getAllByTestId('language-button-option');
  await user.click(options[0]);
  const languageFromLocalStorage = localStorage.getItem('language');
  expect(languageFromLocalStorage).toBe('en');
  expect(options[0].getElementsByTagName('h3')[0]).toHaveTextContent('English');
  expect(options[0].getElementsByTagName('img')[0]).toHaveAttribute(
    'src',
    UsFlag
  );
});

test('option 1 OnClick Function shloud set the language (en) into localStorage', async () => {
  const user = userEvent.setup();
  render(<LanguageButton />);

  await user.click(screen.getByTestId('language-botton-toggle'));

  const options = screen.getAllByTestId('language-button-option');
  await user.click(options[0]);
  const languageFromLocalStorage = localStorage.getItem('language');
  expect(languageFromLocalStorage).toBe('en');
});

test('option 2 shloud has Arabic title and Iraq Flag', async () => {
  const user = userEvent.setup();
  render(<LanguageButton />);

  await user.click(screen.getByTestId('language-botton-toggle'));

  const options = screen.getAllByTestId('language-button-option');
  expect(options[1].getElementsByTagName('h3')[0]).toHaveTextContent('Arabic');
  expect(options[1].getElementsByTagName('img')[0]).toHaveAttribute(
    'src',
    IraqFlag
  );
});

test('option 2 OnClick Function shloud set the language (ar) into localStorage', async () => {
  const user = userEvent.setup();
  render(<LanguageButton />);

  await user.click(screen.getByTestId('language-botton-toggle'));

  const options = screen.getAllByTestId('language-button-option');
  await user.click(options[1]);
  const languageFromLocalStorage = localStorage.getItem('language');
  expect(languageFromLocalStorage).toBe('ar');
});
