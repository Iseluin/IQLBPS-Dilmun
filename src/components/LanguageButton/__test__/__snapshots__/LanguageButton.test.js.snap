// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguageButton component  1`] = `
<div
  class="relative inline-block text-left"
  data-headlessui-state=""
  data-testid="language-button"
>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="md:ml-6 text-2xl md:mt-0 mt-2 text-background hover:text-secondary duration-500"
      data-headlessui-state=""
      data-testid="language-botton-toggle"
      id="headlessui-menu-button-:r0:"
      type="button"
    >
      <img
        alt="English"
        class="h-8 w-8"
        src="us-flag.svg"
      />
    </button>
  </div>
</div>
`;
